### PR TITLE
CORENRN_ENABLE_LEGACY_UNITS default OFF

### PR DESCRIPTION
**Pull request by nrnhines**
Consistency of units selection for NEURON and CoreNEURON checked
 Modern units comes from copy of nrnunits-modern.h
 eion.cpp nrn_nernst and nrn_ghk computes according to the units selection.

Partially responds to #405 and neuronsimulator/nrn#521
Interacts with neuronsimulator/nrn#793 and BlueBrain/mod2c#53

### CODE REVIEW

File: coreneuron/apps/main1.cpp [hunk 1]
```diff
@@ -70,6 +70,11 @@ const char* corenrn_version() {
     return coreneuron::bbcore_write_version;
 }
 
+// the CORENRN_LegacyFR determined by CORENRN_ENABLE_LEGACY_UNITS
+int corenrn_units_use_legacy() {
```
Comment (alkino) [inline coreneuron/apps/main1.cpp:74]: bool corenrn_units_use_legacy()

File: coreneuron/io/global_vars.cpp [hunk 2]
```diff
@@ -32,6 +33,16 @@ void hoc_register_var(DoubScal* ds, DoubVec* dv, VoidFunc*) {
     }
 }
 
+// Only checked here in file mode. Already checked via corenrn_units_use_legacy
+// in direct mode since in the latter case we do not want to abort if
+// inconsistent.
+static void chk_nrnunit_consist(int b) {
+    if (b != CORENRN_LegacyFR) {
```
Comment (alkino) [inline coreneuron/io/global_vars.cpp:40]: Is it possible to use the function above?
And if the function can be a boolean with a better name, I prefer.

Comment (nrnhines) [reply coreneuron/io/global_vars.cpp:40]: Changed  ```CORENRN_LegacyFR``` to ```CORENRN_UseLegacyUnits```

File: coreneuron/apps/main1.cpp (hunk 1 above)
Comment (nrnhines) [reply coreneuron/apps/main1.cpp]: now returns bool

File: coreneuron/io/global_vars.cpp (hunk 2 above)
Comment (alkino) [reply coreneuron/io/global_vars.cpp]: Sorry, I mean the **argument** can be a boolean and have a better name!

Comment (nrnhines) [reply coreneuron/io/global_vars.cpp]: Inlined the function

**Comment by pramodk** [comment]
* Jenkins Pipeline and Simulation Stack is going to fail because of `bbcore_version` is changed and also units default is changed.
* Travis CI is not going to run fast because of Travis CI backlog.

Because of this I will go ahead with the merge of this PR. We will sychronise branches later for CI.

**Review by pramodk — APPROVED** [review]
LGTM

**Pull request merged**